tests/unit_tests/test_dqc_us_0004.py

```diff
@@ -36,12 +40,12 @@ def test_bv_errors(self, patched_decimals):
             liabilities_concept.qname: [liabilities_fact]
         }
 
-        modelXbrl = Mock()
-        modelXbrl.nameConcepts = mock_name_concepts_dict
-        modelXbrl.factsByQname = mock_facts_by_qname
+        model_xbrl = Mock(spec='arelle.modelXbrl.ModelXbrl')
```

Comment:
I thought you had to spec with an actual object reference, but here you're specing with a string to the object... are you sure that works?